### ISSUE TITLE
Some costumes only display in the paint editor and not on the stage

### ISSUE THREAD

**Issue by kchadha** [LLK/scratch-gui#4435]
### Expected Behavior

Costumes should display in the stage.

### Actual Behavior

In the following projects, animations look really blinky (this is not how the project runs in scratch 2.0). Looking inside, it looks like many of the costumes are not rendering on the stage at all:

#82443924 - takes a long time to load, Sprite1 and Sprite2 manifest the issue; the following is a sprite with fewer costumes (costumes 1 - 18 are broken in this sprite among others); rename to .sprite3

[broken-costumes.zip](https://github.com/LLK/scratch-gui/files/2798487/broken-costumes.zip)

#244084125 - Sprite3 and Sprite5 in this project manifest the issue

**Comment by griffpatch** [comment]
I have experienced this on numerous projects, usually vector costumes,
where one costume appears black in the thumbnail of the costume editor and
on stage, bit appears fine in the paint editor itself. By simply moving the
costume slightly in the paint editor it fixed itself.

**Comment by cwillisf** [comment]
TL;DR: I think this is still `needs-repro`.

Both projects load and run correctly for me with no blinking or flickering during the animation. All frames of animation seem to be present.

The `broken-costumes.zip` sprite seems to be missing a bunch of costumes: the costume names go `costume1`, `costume2`, ..., `costume26`, `costume322`, `costume323`, skipping costumes 27-321. That's consistent with the JSON and SVGs contained within the ZIP, though, so Scratch 3.0 is loading the data accurately. I guess the problem happened before exporting the sprite and the export just faithfully preserved the broken state?

If anyone (@kchadha?) can still get the broken behavior on project 82443924 or 244084125 then this might be specific to certain configurations. If not, maybe this issue or those projects have been fixed within the last 4 days.

**Comment by paulkaplan** [comment]
@cwillisf i also could not repro on chrome 71. But using chrome 69, i was able to get some costumes to not show up on the stage. I noticed these projects have a ton of costumes, I'm wondering if it is a video memory/texture loading/webgl problem, since downloading the costumes that do not show up and re-uploading them to a different project display just fine.

**Comment by paulkaplan** [comment]
@griffpatch did you mean "black" or "blank". I'm seeing the issue (on chrome 69) in the project @kchadha gave where the costume is _blank_ on the stage only (appears fine in costume editor, costume tile and sprite tile). But if you are seeing the costume appear _black_ in the thumbnails/stage, that sounds like a different (and bad!) issue. Could you give a project link where you've seen this, as well as what browser/version/OS you saw it with?

**Comment by griffpatch** [comment]
I have a repo for this... It occurs with a single perfectly horizontal line in the vector editor.
1) Just start a new costume
2) choose the line tool
3) Hold down 'shift' and drag a line across the screen.

At this point the costume will be displaying in the editor, but not in the project or thumbnail
Adding a second line will make both lines visible (even if horizontal), and removing it again will make it invisible.

I suspect that the height of the costume is being calculated as 0 even if the line is very thick? This is then causing the image to not be displayed...

Am I right in thinking the issue of counting thick lines in a costumes dimensions might have been addressed in another issue/log too?

**Comment by paulkaplan** [comment]
@griffpatch thanks for the repro! I created a new issue to track that specific problem, since that is one that can be easily repro'd. https://github.com/LLK/scratch-gui/issues/4479

**Comment by BryceLTaylor** [comment]
I'm closing this, since the initial issue seemed to be related to a specific (and now old) version of chrome and is no longer a problem and the rest have been moved into a new issue.
